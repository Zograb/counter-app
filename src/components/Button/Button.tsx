import { ComponentPropsWithoutRef, FC } from 'react';
import clsx from 'clsx';

export const Button: FC<ComponentPropsWithoutRef<'button'>> = ({ className, ...otherProps }) => (
  <button
    className={clsx(
      'h-10 px-6 text-3xl rounded bg-neutral',
      'transition-all duration-200',
      'hover:bg-neutral-hover active:scale-90 active:opacity-80',
      className,
    )}
    {...otherProps}
  />
);
