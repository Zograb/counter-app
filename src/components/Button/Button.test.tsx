import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import { Button } from './Button';

describe('button', () => {
  it('should match snapshot', () => {
    const { container } = render(<Button>BUTTON</Button>);
    expect(container).toMatchSnapshot();
  });

  it('should render button component', () => {
    const { getByRole } = render(<Button>BUTTON</Button>);
    const button =  getByRole('button');
    expect(button).toBeInTheDocument();
  });

  it('should call onClick on button click', () => {
    const clickFn = jest.fn();
    const { getByRole } = render(<Button onClick={clickFn}>BUTTON</Button>);
    const button = getByRole('button');
    button.click();
    expect(clickFn).toHaveBeenCalled();
  });
});
