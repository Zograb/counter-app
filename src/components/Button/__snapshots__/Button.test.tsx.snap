// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button should match snapshot 1`] = `
<div>
  <button
    class="h-10 px-6 text-3xl rounded bg-neutral transition-all duration-200 hover:bg-neutral-hover active:scale-90 active:opacity-80"
  >
    BUTTON
  </button>
</div>
`;
